spec/controllers/registrations_controller_spec.rb

```diff
@@ -50,7 +50,10 @@
     context "with invalid params" do
       context "empty fields" do
         before(:each) do
-          post :create, user: { name: "", email: "", password: "", password_confirmation: "", gender: "male" }
+          post :create,
+            params: {
+              user: { name: "", email: "", password: "", password_confirmation: "", gender: "male" }
```

Comment:
You can define a function that has the data as entry and returns the params hash. Then use this function everywhere with the data you need.